src/test/java/uk/gov/hmcts/reform/sscs/service/LetterUtilsTest.java

```diff
@@ -415,4 +430,250 @@ private String getExpectedName(final String otherPartyId, final NotificationWrap
                 .orElse(""));
     }
 
+    private SscsCaseData setupTestData(DynamicList sender) {
+        return SscsCaseData.builder()
+                .originalSender(sender)
+                .jointParty(JOINT_PARTY_WITH_ID)
+                .otherParties(buildOtherPartyData())
+                .appeal(Appeal.builder()
+                    .appellant(APPELLANT_WITH_ID)
+                    .rep(REPRESENTATIVE_WITH_ID)
+                    .build())
+                .build();
+    }
+
+    private List<CcdValue<OtherParty>> buildOtherPartyData() {
+        return List.of(CcdValue.<OtherParty>builder()
+            .value(OtherParty.builder()
+                .id("OP123456")
+                .name(Name.builder()
+                    .firstName("Other")
+                    .lastName("Party")
+                    .build())
+                .otherPartySubscription(Subscription.builder().email("other@party").subscribeEmail("Yes").build())
+                .rep(Representative.builder()
+                    .id("OPREP123456")
+                    .name(Name.builder()
+                        .firstName("OtherParty")
+                        .lastName("Representative")
+                        .build())
+                    .hasRepresentative(YesNo.YES.getValue())
+                    .build())
+                .build())
+            .build(),
+            CcdValue.<OtherParty>builder()
+                .value(OtherParty.builder()
+                    .id("OP7890123")
+                    .name(Name.builder()
+                        .firstName("Other 1")
+                        .lastName("Party 1")
+                        .build())
+                    .otherPartySubscription(Subscription.builder().email("other@party").subscribeEmail("Yes").build())
+                    .rep(Representative.builder()
+                        .hasRepresentative(YesNo.NO.getValue())
+                        .build())
+                    .build())
+                .build()
+            );
+    }
```

Comment:
Likewise, global var setup in before each. In tests it is almost always preferrable to avoid helper methods for setting up data. Otherwise you might need to test those methods that they do what you expect.

Reply:
This data setup is not used for all the test cases. It is specific for few test case only.

Reply:
This method is called by the @Before setup method.